### PR TITLE
Docs for client properties

### PR DESCRIPTION
**Pull request by sophokles73**

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java [hunk 1]
```diff
@@ -21,13 +21,26 @@
...
     private String name;
     private String host = "localhost";
     private int port = Constants.PORT_AMQPS;
     private String username;
     private char[] password;
     private String amqpHostname;
-    private long waitMillisForCredits = 10L;
+    private long flowLatency = 10L;
     private int initialCredits = 1000;
     private long requestTimeoutMillis = 200L;
```
Comment (pellmann) [inline core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: I think your newly created static defaults should be assigned here for flowLatency, initialCredits and requestTimeoutMillis.

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java [hunk 2]
```diff
@@ -21,13 +21,26 @@
...
 public class ClientConfigProperties extends AbstractConfig {
 
+    /**
+     * The default amount of time to wait for credits after link creation.
+     */
+    public static final long DEFAULT_FLOW_LATENCY = 20L; //ms
+    /**
+     * The default number of credits issued by the receiver side of a link.
+     */
+    public static final int  DEFAULT_INITIAL_CREDITS = 200;
```
Comment (pellmann) [inline core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: Did you changed this on purpose? If yes, do think 200 is a better value?

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java [hunk 3]
```diff
@@ -21,13 +21,26 @@
  */
 public class ClientConfigProperties extends AbstractConfig {
 
+    /**
+     * The default amount of time to wait for credits after link creation.
+     */
+    public static final long DEFAULT_FLOW_LATENCY = 20L; //ms
```
Comment (pellmann) [inline core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: Did you changed this on purpose? If yes, do think 20 is a better value?

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java (hunk 1 above)
Comment (sophokles73) [reply core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: absolutely, I simply forgot to do that. Thanks for spotting :+1:

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java (hunk 3 above)
Comment (sophokles73) [reply core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: Running locally on my machine I still experienced 503 results using 10ms. Do you think 20ms is too much for a default?

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java (hunk 2 above)
Comment (sophokles73) [reply core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: I thought 200 would be a little more *defensive* and *cautious*. I also have been inspired by the 250 credits that the dispatch router hands out by default (and which seem to be enough to be able to process 30000 msg/s).

File: core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java (hunk 3 above)
Comment (pellmann) [reply core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java]: No, I was only interested in your thoughts - on my machine 10ms worked, but than the default of 20ms might be better.

**Comment by sophokles73** [comment]
@pellmann 
WDYT?

**Pull request merged**